### PR TITLE
Expose streams in binaryop APIs

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Contributes to #925

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/streams/binaryop_test.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+{
+  cudf::numeric_scalar<int32_t> lhs{42, true, cudf::test::get_default_stream()};
+  cudf::test::fixed_width_column_wrapper<int32_t> rhs{15, 25, 35, 45, 55};
+
+  cudf::binary_operation(lhs,
+                         rhs,
+                         cudf::binary_operator::ADD,
+                         cudf::data_type(cudf::type_to_id<int32_t>()),
+                         cudf::test::get_default_stream());
+}
```
Comment (karthikeyann) [inline cpp/tests/streams/binaryop_test.cpp:61]: is udf test (ptx) not required?

Comment (vyasr) [reply cpp/tests/streams/binaryop_test.cpp:61]: Ah good catch I totally forgot that API here. I'll add a test.

Comment (vyasr) [reply cpp/tests/streams/binaryop_test.cpp:61]: Done.

**Review by ttnghia — APPROVED** [review]

**Pull request merged**